integrationTests/ts/package.json

```diff
@@ -6,11 +6,18 @@
   },
   "dependencies": {
     "graphql": "file:../graphql.tgz",
-    "typescript-4.1": "npm:typescript@4.1.x",
```

Comment:
These changes break TS 4.1, but that would probably be okay for a new major of `graphql`.

Reply:
I was going with the [DefinitelyTyped support window for TypeScript versions](https://github.com/DefinitelyTyped/DefinitelyTyped?tab=readme-ov-file#support-window) here.

No other reason to drop them than that not even Microsoft themself want to maintain types for those old versions any more - it seemed reasonable as I had to touch this already, and we won't be able to drop any TS versions between majors, so if we even want to remotely consider using modern features in the future, this would be the time.

Reply:
As for 4.1, it doesn't resolve the `exports`, while all other versions do.

Tons of
```
basic-test.ts:1:38 - error TS7016: Could not find a declaration file for module 'graphql/execution'. '/private/var/folders/ms/w25rss0n4vq830bvptjjkqvr0000gp/T/graphql-js-integrationTmp/ts/node_modules/graphql/execution/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/graphql` if it exists or add a new declaration (.d.ts) file containing `declare module 'graphql/execution';`

1 import type { ExecutionResult } from 'graphql/execution';
                                       ~~~~~~~~~~~~~~~~~~~

basic-test.ts:3:29 - error TS7016: Could not find a declaration file for module 'graphql'. '/private/var/folders/ms/w25rss0n4vq830bvptjjkqvr0000gp/T/graphql-js-integrationTmp/ts/node_modules/graphql/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/graphql` if it exists or add a new declaration (.d.ts) file containing `declare module 'graphql';`

3 import { graphqlSync } from 'graphql';
                              ~~~~~~~~~

basic-test.ts:4:65 - error TS7016: Could not find a declaration file for module 'graphql/type'. '/private/var/folders/ms/w25rss0n4vq830bvptjjkqvr0000gp/T/graphql-js-integrationTmp/ts/node_modules/graphql/type/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/graphql` if it exists or add a new declaration (.d.ts) file containing `declare module 'graphql/type';`

4 import { GraphQLString, GraphQLSchema, GraphQLObjectType } from 'graphql/type';
                                                                  ~~~~~~~~~~~~~~
```

etc.

Reply:
Ah. Then why don’t we remove support in a separate PR so it goes in as a separate breaking change